oauthproxy_test.go

```diff
@@ -156,6 +156,32 @@ func TestEncodedSlashes(t *testing.T) {
 	}
 }
 
+func TestNewReverseProxyWithTimeOut(t *testing.T) {
+	opts := NewOptions()
+	opts.Timeout = 1 * time.Millisecond
+
+	backend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
upstream?

Reply:
This was taken almost as-is from an existing older test. I will fix this in the follow-up.

Reply:
I kept the name as **backend** for now as the rest of the test was using **fronted**. Unless you prefer **upstream** and **downstream**?

Reply:
no problems there, that was a mere nit